.ci/integrationTestEC.groovy

```diff
@@ -0,0 +1,176 @@
+        }
+        axes {
+          axis {
+              name 'TEST'
+              values 'all', 'dotnet', 'go', 'java', 'nodejs', 'python', 'ruby', 'rum'
+          }
+          axis {
+              name 'ELASTIC_STACK_VERSION'
+              values '8.0.0-SNAPSHOT', '7.7.0-SNAPSHOT', '7.6.1-SNAPSHOT', '6.8.7-SNAPSHOT'
+          }
+        }
+        stages {
+          stage('Prepare Test'){
+            steps {
+              log(level: "INFO", text: "Running tests - ${ELASTIC_STACK_VERSION} x ${TEST}")
+              deleteDir()
+              unstash 'source'
+            }
+          }
+          stage('Provision Elastic Cloud environment'){
+            steps {
+              dockerLogin(secret: "${DOCKERELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
+              dir("${EC_DIR}/ansible"){
+                withVaultEnv(){
+                  sh(label: "Deploy Cluster", script: "make deploy-cluster")
+                  //sh(label: "Rename cluster-info folder", script: "mv cluster-info cluster-info-${ELASTIC_STACK_VERSION}x${TEST}")
+                  //archiveArtifacts(allowEmptyArchive: true, artifacts: 'cluster-info*/**')
+                }
+              }
+            }
+          }
+          stage("Test") {
+            steps {
+              dir("${BASE_DIR}"){
+                withConfigEnv(){
+                  sh ".ci/scripts/${TEST}.sh"
+                }
+              }
+            }
+          }
+        }
+        post {
+          cleanup {
+            wrappingUp("${TEST}")
+            destroyClusters()
+          }
+        }
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def withVaultEnv(Closure body){
+  getVaultSecret.readSecretWrapper {
+    def token = getVaultSecret.getVaultToken(env.VAULT_ADDR, env.VAULT_ROLE_ID, env.VAULT_SECRET_ID)
+    withEnvMask(vars: [
+      [var: 'VAULT_TOKEN', password: token],
+      [var: 'VAULT_AUTH_METHOD', password: 'approle'],
+      [var: 'VAULT_AUTHTYPE', password: 'approle']
+    ]){
+      body()
+    }
+  }
+}
+
+def withConfigEnv(Closure body) {
+  def config = readYaml(file: "ansible/config-general.yml")
+  def apm = getVaultSecret(secret: "secret/${config.k8s_vault_apm_def_secret}")?.data
+  def es = getVaultSecret(secret: "secret/${config.k8s_vault_elasticsearch_def_secret}")?.data
+  def kb = getVaultSecret(secret: "secret/${config.k8s_vault_apm_def_secret}")?.data
+
+  withEnvMask(vars: [
+    [var: 'APM_SERVER_URL', password: apm.value.url],
+    [var: 'APM_SERVER_SECRET_TOKEN', password: apm.value.token],
+    [var: 'ES_URL', password: es.value.url],
+    [var: 'ES_USER', password: es.value.username],
+    [var: 'ES_PASS', password: es.value.password],
+    [var: 'KIBANA_URL', password: kb.value.url],
+    [var: 'BUILD_OPTS', password: "${params.BUILD_OPTS} --apm-server-url ${apm.value.url} --apm-server-secret-token ${apm.value.token}"]
+  ]){
+    body()
+  }
+}
+
+def wrappingUp(label){
```

Comment:
Tiny nit: could we rename this method using a more specific name? Something like `wrappingUpBuildArtefacts` or `gatherBuildArtefacts`... 🤔 Otherwise, we have to read the implementation to understand its behavior